tuned/plugins/plugin_scheduler.py

```diff
@@ -350,8 +410,103 @@ def _convert_sched_cfg(self, vals):
+				no_error = True)):
+					log.error("Unable to initialize 'cpuset.cpus' for cgroup '%s'" % path)
+
+	def _cgroup_initialize_groups(self):
+		if self._cgroup is not None and not self._cgroup in self._cgroups:
+			self._cgroup_create_group(self._cgroup)
+		for cg in self._cgroups:
+			self._cgroup_create_group(cg)
+
+	def _cgroup_initialize(self):
+		log.debug("Initializing cgroups settings")
+		try:
+			os.makedirs(self._cgroup_mount_point, consts.DEF_CGROUP_MODE)
+		except OSError as e:
+			log.error("Unable to create cgroup mount point: %s" % e)
+		(ret, out) = self._cmd.execute(["mount", "-t", "cgroup", "-o", "cpuset", "cpuset", self._cgroup_mount_point])
+		if ret != 0:
+			log.error("Unable to mount '%s'" % self._cgroup_mount_point)
+
+	def _remove_dir(self, cgroup):
+		try:
+			os.rmdir(cgroup)
+		except OSError as e:
+			log.error("Unable to remove directory '%s': %s" % (cgroup, e))
+
+	def _cgroup_finalize_groups(self):
+		if self._cgroup is not None and not self._cgroup in self._cgroups:
+			self._remove_dir("%s/%s" % (self._cgroup_mount_point, self._cgroup))
+		for cg in self._cgroups:
+			self._remove_dir("%s/%s" % (self._cgroup_mount_point, cg))
+
+	def _cgroup_finalize(self):
+		log.debug("Removing cgroups settings")
+		(ret, out) = self._cmd.execute(["umount", self._cgroup_mount_point])
+		if ret != 0:
+			log.error("Unable to umount '%s'" % self._cgroup_mount_point)
+			return False
+		self._remove_dir(self._cgroup_mount_point)
+		d = os.path.dirname(self._cgroup_mount_point)
+		if (d != "/"):
+			self._remove_dir(d)
+
+	def _cgroup_set_affinity_one(self, cgroup, affinity, backup = False):
+		log.debug("Setting cgroup '%s' affinity to '%s'" % (cgroup, affinity))
+		path = "%s/%s/%s" % (self._cgroup_mount_point, cgroup, "cpuset.cpus")
+		if backup:
+			orig_affinity = self._cmd.read_file(path, err_ret = "ERR", no_error = True)
+			if orig_affinity != "ERR":
+				self._cgroups_original_affinity[cgroup] = orig_affinity
+			else:
+				log.warn("Unable to save affinity of cgroup '%s'" % cgroup)
+		if not self._cmd.write_to_file(path, affinity, no_error = True):
+			log.error("Unable to set affinity '%s' for cgroup '%s'" % (affinity, cgroup))
+
+	def _cgroup_set_affinity(self):
+		log.debug("Setting cgroups affinities")
+		if self._affinity is not None and self._cgroup is not None and not self._cgroup in self._cgroups:
+			self._cgroup_set_affinity_one(self._cgroup, self._affinity, backup = True)
+		for cg in self._cgroups.items():
+			self._cgroup_set_affinity_one(cg[0], cg[1], backup = True)
+
+	def _cgroup_restore_affinity(self):
+		log.debug("Restoring cgroups affinities")
+		for cg in self._cgroups_original_affinity.items():
+			self._cgroup_set_affinity_one(cg[0], cg[1])
+
 	def _instance_apply_static(self, instance):
+		# need to get "cgroup_mount_point_init", "cgroup_mount_point", "cgroup_groups_init",
+		# "cgroup", and initialize mount point and cgroups before super class init
```

Comment:
Did you mean "before calling the _instance_apply_static method of the super class"? I understand "init" to mean `__init__`.

Reply:
Yep, typo.